libs/core/tests/unit_tests/test_tools.py

```diff
@@ -483,6 +483,102 @@ def test_structured_tool_lambda_multi_args_schema() -> None:
     assert tool.args == expected_args
 
 
+def test_structured_tool_from_function_partial_docstring() -> None:
+    """Test that structured tools can be created from partial functions."""
+
+    def foo(bar: int, baz: str) -> str:
+        """Docstring
+
+        Args:
+            bar: the bar value
+            baz: the baz value
+        """
+        raise NotImplementedError
+
+    structured_tool = StructuredTool.from_function(func=partial(foo, baz="foo"))
+    assert structured_tool.name == "foo"
+    assert structured_tool.args == {
+        "bar": {"title": "Bar", "type": "integer"},
+        "baz": {"title": "Baz", "type": "string"},
```

Comment:
don't we want baz to not appear as an  arg since its already been set?

Reply:
I'm looking to assign the same tool to several agents, ensuring that some of the parameters are preset, whereas others are dynamically determined by a large model each time an agent uses this tool.

Also, I found similar test scenarios in the test cases; unfortunately, right now, the `StructuredTool.from_function` method doesn't support them yet.

https://github.com/langchain-ai/langchain/blob/master/libs/core/tests/unit_tests/test_tools.py#L486

Reply:
> don't we want baz to not appear as an arg since its already been set?

Hi, just checking in on this PR. Any feedback? Thanks!

Reply:
@coolbeevip I think some more work is needed on this PR. IMO it doesn't make sense for the args schema of the tool to still retain the partialed-out parameter. In your first two tests, you can pass the test by just setting `func=foo`.

We document some other ways to do this in [this guide](https://python.langchain.com/docs/how_to/tool_runtime/) as well.